pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -634,7 +634,9 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
      * the `amounts` array.
      */
     function _upscaleArray(uint256[] memory amounts, uint256[] memory scalingFactors) internal pure {
-        uint256 length = Math.max(amounts.length, scalingFactors.length);
+        InputHelpers.ensureInputLengthMatch(amounts.length, scalingFactors.length);
+
+        uint256 length = amounts.length;
         for (uint256 i = 0; i < length; ++i) {
```

Comment:
```suggestion
        uint256 numTokens = amounts.length;
        InputHelpers.ensureInputLengthMatch(numTokens, scalingFactors.length);

        for (uint256 i = 0; i < numTokens; ++i) {
```
If we want to avoid calling length twice on amounts.